nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -1551,6 +1551,67 @@ public void OptionalEmptinessUncheckedTest() {
         .doTest();
   }
 
+  @Test
+  public void OptionalEmptinessRxPositiveTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber,io.reactivex",
+                "-XepOpt:NullAway:UnannotatedSubPackages=com.uber.lib.unannotated",
+                "-XepOpt:NullAway:CheckOptionalEmptiness=true"))
+        .addSourceLines(
+            "TestPositive.java",
+            "package com.uber;",
+            "import java.util.Optional;",
+            "import io.reactivex.Observable;",
+            "public class TestPositive {",
+            "  private static boolean perhaps() { return Math.random() > 0.5; }",
+            "  void foo(Observable<Optional<String>> observable) {",
+            "     observable",
+            "           .filter(optional -> optional.isPresent() || perhaps())",
+            "           // BUG: Diagnostic contains: Optional optional can be empty",
+            "           .map(optional -> optional.get().toString());",
+            "     observable",
+            "           .filter(optional -> optional.isPresent() || perhaps())",
+            "           // BUG: Diagnostic contains: returning @Nullable expression from method with @NonNull",
```

Comment:
This error message seems inappropriate, so need to make some changes in the code to have the correct error message.

Reply:
See comments on #307 and re-add this one plus that fix (assuming it works, of course)

Reply:
Done 👍